lib/pub_sub/message.rb

```diff
@@ -11,20 +11,22 @@ def initialize(payload)
     end
 
     def process
-      validate_message!
+      begin
+        validate_message!
+      rescue PubSub::ServiceUnknown => e
+        return
+      end
       handler.process(data)
     end
 
     def validate_message!
       messages = PubSub.config.subscriptions[sender]
-      if messages.nil?
-        warning = "WARN: We received a message from #{sender} but we do " \
-                'not subscribe to that service.'
-        PubSub.logger.warn(warning)
-      end
-      unless messages.include?(type)
-        error = "We received a message from #{sender} but it was " \
-                "of unknown type #{type}."
+      if messages.blank?
+        warning = "#{PubSub.config.service_name} received a message from #{sender} but no matching subscription exists for that sender"
```

Comment:
I am not convinced. The line breaks your editor produces are there exactly for that purpose. Not only do they look quite fine, but adding extra syntax fluff only to keep the 80-char-per-line terminals from the 90s happy is silly